cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -431,13 +439,14 @@ private Future<Void> reconcile(Reconciliation reconciliation, String host,
                                                 kafkaRebalance.getMetadata().getName(), desiredStatusAndMap.getLoadMap())
                                                 .compose(i -> updateStatus(reconciliation, currentKafkaRebalance, desiredStatusAndMap.getStatus(), null))
                                                 .compose(updatedKafkaRebalance -> {
-                                                    String message = "State updated to [{}] ";
-                                                    if (rawRebalanceAnnotation(updatedKafkaRebalance) == null) {
-                                                        LOGGER.infoCr(reconciliation, message + "and annotation {} is not set ",
-                                                                rebalanceStateConditionType(updatedKafkaRebalance.getStatus()),
-                                                                ANNO_STRIMZI_IO_REBALANCE);
-                                                    } else {
-                                                        LOGGER.infoCr(reconciliation, message + "with annotation {}={} ",
+                                                    String message = "";
+                                                    if (currentKafkaRebalance.getStatus() != null
+                                                            && updatedKafkaRebalance.getStatus() != null
+                                                            && !Objects.equals(rebalanceStateConditionType(currentKafkaRebalance.getStatus()), rebalanceStateConditionType(updatedKafkaRebalance.getStatus()))) {
```

Comment:
As per Jakub's comment before I think a traditional `equals` on one of the results of the `rebalanceStateConditionType` method is better (in the end it provides a String).